src/main/java/edu/cornell/kfs/concur/batch/service/impl/ConcurRequestExtractFileServiceImpl.java

```diff
@@ -14,38 +13,89 @@
+            }
+            requestExtractFile.setFileName(parseRequestExtractFileNameFrom(requestExtractFullyQualifiedFileName));
+            processingSuccessful = getConcurCashAdvancePdpFeedFileService().createPdpFeedFileForValidatedDetailFileLines(requestExtractFile);
+            if (processingSuccessful) {
+                try {
+                    getConcurCashAdvancePdpFeedFileService().createDoneFileForPdpFile(requestExtractFile.getFullyQualifiedPdpFileName());
+                } catch (IOException ioe) {
+                    LOG.error("ConcurCashAdvancePdpFeedFileService().createDoneFileForPdpFile generated IOException attempting to create .done file for generated PdpFeedFile: " + requestExtractFile.getFullyQualifiedPdpFileName());
+                    processingSuccessful = false;
+                } catch (FileStorageException fse) {
+                    LOG.error("ConcurCashAdvancePdpFeedFileService().createDoneFileForPdpFile generated FileStorageException attempting to create .done file for generated PdpFeedFile: " + requestExtractFile.getFullyQualifiedPdpFileName());
+                    processingSuccessful = false;
+                }
+            }
+        }
+        LOG.debug("method processFile:: requestExtractFile data after processing: " + KFSConstants.NEWLINE + requestExtractFile.toString());
+        return processingSuccessful;
+    }
+
+    private ConcurRequestExtractFile loadFileIntoParsedDataObject(String requestExtractFullyQualifiedFileName) {
+        Object parsedFile = getConcurBatchUtilityService().loadFile(requestExtractFullyQualifiedFileName, getBatchInputFileType());
         List<ConcurRequestExtractFile> requestExtractFiles = (ArrayList<ConcurRequestExtractFile>) parsedFile;
-        processingSuccess = getConcurRequestExtractFileValidationService().requestExtractHeaderRowValidatesToFileContents(requestExtractFiles);
-        return processingSuccess;
+        if (requestExtractFiles.size() != 1) {
+            LOG.error("Single physical file " + requestExtractFullyQualifiedFileName + " should have translated into single parsed file. More or less than one parsed file was detected.");
+            throw new RuntimeException("Single physical file " + requestExtractFullyQualifiedFileName + " should have translated into single parsed file. More or less than one parsed file was detected.");
+        }
+        else {
+            ConcurRequestExtractFile requestExtractFile = requestExtractFiles.get(0);
+            return requestExtractFile;
+        }
     }
 
-    public void setBatchInputFileService(BatchInputFileService batchInputFileService) {
-        this.batchInputFileService = batchInputFileService;
+    private String parseRequestExtractFileNameFrom(String requestExtractFullyQualifiedFileName) {
+        //plus 1 because we do not want the forward slash
+        return requestExtractFullyQualifiedFileName.substring((requestExtractFullyQualifiedFileName.lastIndexOf(ConcurConstants.FORWARD_SLASH)) + 1);
```

Comment:
You could alternatively use StringUtils.substringAfterLast() here, which would be a little cleaner.